### PR TITLE
KT-24425 wrong hint remove redundant Companion

### PR DESCRIPTION
**Pull request by t-kameyama**
https://youtrack.jetbrains.com/issue/KT-24425

### CODE REVIEW

File: idea/src/org/jetbrains/kotlin/idea/inspections/RedundantCompanionReferenceInspection.kt [hunk 1]
```diff
@@ -29,6 +32,21 @@ class RedundantCompanionReferenceInspection : AbstractKotlinInspection() {
             if (!objectDeclaration.isCompanion()) return
             if (expression.text != objectDeclaration.name) return
 
+            val property = parent.getStrictParentOfType<KtProperty>()
+            if (property != null && property.name == parent.selectorExpression?.text) return
```
Comment (mglukhikh) [inline idea/src/org/jetbrains/kotlin/idea/inspections/RedundantCompanionReferenceInspection.kt]: What about case like below? Here we still cannot delete `Companion`, but your check does not work here.

```
class My {
    val name = "My name"
    val greeting = "Hello " + Companion.name
    companion object {
        val name = "Your name"
    }
}
```

File: idea/src/org/jetbrains/kotlin/idea/inspections/RedundantCompanionReferenceInspection.kt [hunk 2]
```diff
@@ -29,6 +32,21 @@ class RedundantCompanionReferenceInspection : AbstractKotlinInspection() {
...
+
+            val function = parent.getStrictParentOfType<KtNamedFunction>()
+            if (function != null) {
+                val callee = (parent.selectorExpression as? KtCallExpression)?.calleeExpression
+                if (function.name == callee?.text) {
```
Comment (mglukhikh) [inline idea/src/org/jetbrains/kotlin/idea/inspections/RedundantCompanionReferenceInspection.kt]: Here we can build same example as for properties above

**Comment by t-kameyama** [comment]
Fixed

**Comment by mglukhikh** [comment]
Merged

**Pull request closed without merging**